.github/workflows/ci.yml

```diff
@@ -179,3 +181,47 @@ jobs:
         run: |
           cd $BUILD_DIR/scripts/code_generation/fakegen
           valgrind --tool=memcheck --leak-check=yes ./fakegen_test
+  
+  generate-dbc:
+      name: Generate CAN DBC file
+      runs-on: ubuntu-22.04
+      container:
+        image: ubcformulaelectric/environment:latest
+        credentials:
+          username: ubcformulaelectric
+          password: ${{ secrets.docker_token }}
+      steps:
+        - name: Checkout repository
+          uses: actions/checkout@v2
+          with: # TODO check if submodules are required (can be optimized away) (From bin-dbc-gen)
+            submodules: recursive
+        - name: Generate DBC
+          run: |
+            python3 scripts/code_generation/jsoncan/generate_can_from_json.py --only_dbc --can_data_dir "can_bus/json" --dbc_output "can_bus/dbcs/CanMsgs.dbc"
+        - name: Upload DBC to Artifact
+          uses: actions/upload-artifact@v2
+          with:
+            name: CanMsgs.dbc
+            path: ./can_bus/dbcs/CanMsgs.dbc
```

Comment:
extract this to a new workflow file and reuse it just like we do `build.yml`

Reply:
build.yml is called through a file on the master branch, I can't do this for generate DBC because there is no file on master branch. Thoughts?

Reply:
I pushed a change (i think on my branch) where you can refer to workflow files from your own branch. Check [this post](https://github.blog/changelog/2022-01-25-github-actions-reusable-workflows-can-be-referenced-locally/).

Reply:
ok then, so its fine if the version of DBC gen gets run on the current branch?

Reply:
yes, in fact all of them are ok to run on the current branch. Enforcement of passing is handled by GitHub separately, and cannot be bypassed through local branch action handling/removal